torch/csrc/inductor/aoti_eager/kernel_holder.cpp

```diff
@@ -125,18 +117,19 @@ std::vector<ParameterMetadata> unpack_input_parameters(
       // optional tensor list: std::vector<std::optional<at::Tensor>>
       std::vector<at::Tensor> tensor_list;
       for (const auto& item : stack[idx].toListRef()) {
-        if (item.toOptional<at::Tensor>().has_value()) {
-          tensor_list.push_back(item.toOptional<at::Tensor>().value());
+        auto e = item.toOptional<at::Tensor>();
```

Comment:
This annoyingly adds on extra copy since toOptional was previously an rvalue? No way to check the fullness of the optional on the original item?

Reply:
I know it will complain, but we can add a nolint if it's problematic.

Reply:
No, it created two temporary optional in the old code with one moved. Now it creates one temporary optional that moved conditionally.